mokapot/brew.py

```diff
@@ -194,90 +295,210 @@ def brew(psms, model=None, test_fdr=0.01, folds=3, max_workers=1, rng=None):
             "using the original model."
         )
 
-    LOGGER.info("")
-    res = [
-        p.assign_confidence(s, eval_fdr=test_fdr, desc=d)
-        for p, s, d in zip(psms, scores, descs)
-    ]
-
-    if len(res) == 1:
-        return res[0], models
-
-    return res, models
+    return psms, models, scores, descs
 
 
 # Utility Functions -----------------------------------------------------------
-def _make_train_sets(psms, test_idx):
+def make_train_sets(test_idx, subset_max_train, data_size, rng):
     """
     Parameters
     ----------
-    psms : list of PsmDataset
-        The PsmDataset to get a subset of.
     test_idx : list of list of numpy.ndarray
         The indicies of the test sets
+    subset_max_train : int or None
+        The number of PSMs for training.
+    data_size : list[int]
+        size of the input data
 
     Yields
     ------
     PsmDataset
         The training set.
     """
-    train_set = copy.copy(psms[0])
-    all_idx = [set(range(len(p.data))) for p in psms]
-    for idx in zip(*test_idx):
-        train_set._data = None
-        data = []
-        for i, j, dset in zip(idx, all_idx, psms):
-            data.append(dset.data.loc[list(j - set(i)), :])
-
-        train_set._data = pd.concat(data, ignore_index=True)
-        yield train_set
-
-
-def _predict(dset, test_idx, models, test_fdr):
+    subset_max_train_per_file = []
+    if subset_max_train is not None:
+        subset_max_train_per_file = [
+            subset_max_train // len(data_size) for _ in range(len(data_size))
+        ]
+        subset_max_train_per_file[-1] += subset_max_train - sum(
+            subset_max_train_per_file
+        )
+    chunk_range = 5000000
```

Comment:
Absolutely. Although we are happy to have environment variables to set some of these settings now, we are still not really happy about it. We have issues in our backlog, to change the ENV variables so that they can be better understood by users.

Specifically, we would like to remove ENV variables like "CHUNK_SIZE_X" and replace them with two variables:
"N_THREADS", "GBS_MEMORY" (or something similar) and then derive the different chunk sizes automatically from that. Currently we optimized those variables manually to fit our hardware constraints, which is obviously not optimal.

Would you be ok, with leaving this as is, as we plan to cover this in the upcoming issues?

Reply:
Sounds good!

Reply:
https://github.com/wfondrie/mokapot/pull/119#issuecomment-2231943971 <- Added there to the series of things that should be done but not now